### PR TITLE
doc/dev/cephadm: Note regarding network calls form CLI handlers

### PR DESCRIPTION
**Pull request by sebastian-philipp**
Signed-off-by: Sebastian Wagner <sebastian.wagner@suse.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: doc/dev/cephadm.rst [hunk 1]
```diff
@@ -85,3 +85,18 @@ When you're done, you can tear down the cluster with::
...
+Note regarding network calls form CLI handlers
+==============================================
+
+Executing any cephadm CLI commands like ``ceph orch ls`` will block
+the mon command handler thread within the MGR, thus preventing any
+concurrent CLI calls. Note that pressing ``^C`` will not resolve this
+situation, as *only* the client will be aborted, but not exceution
+itself. This means, cephadm will be completely unresonsive, until the
+execution of the CLI handler is fully completed. Note that even
+`ceph orch ps`` will not respond, while another handler is executed.
```
Comment (jschmid1) [inline doc/dev/cephadm.rst:98]:
```suggestion
``ceph orch ps`` will not respond, while another handler is executed.
```

File: doc/dev/cephadm.rst [hunk 2]
```diff
@@ -85,3 +85,18 @@ When you're done, you can tear down the cluster with::
...
+
+This means, we should only do very few calls to remote hosts synchronously. 
+As a guideline, cephadm should do at most ``O(1)`` networks calls in CLI handlers. 
+Everything else should be one asynchronously in othter threads, like ``serve()``.
```
Comment (sebastian-philipp) [inline doc/dev/cephadm.rst:102]:
```suggestion
Everything else should be done asynchronously in other threads, like ``serve()``.
```

File: doc/dev/cephadm.rst [hunk 3]
```diff
@@ -85,3 +85,18 @@ When you're done, you can tear down the cluster with::
 
   sudo ../src/ckill.sh   # or,
   sudo ../src/cephadm/cephadm rm-cluster --force --fsid `cat fsid`
+
+Note regarding network calls form CLI handlers
```
Comment (mgfritch) [inline doc/dev/cephadm.rst:89]:
```suggestion
Note regarding network calls from CLI handlers
```

File: doc/dev/cephadm.rst [hunk 4]
```diff
@@ -85,3 +85,18 @@ When you're done, you can tear down the cluster with::
...
+Executing any cephadm CLI commands like ``ceph orch ls`` will block
+the mon command handler thread within the MGR, thus preventing any
+concurrent CLI calls. Note that pressing ``^C`` will not resolve this
+situation, as *only* the client will be aborted, but not exceution
+itself. This means, cephadm will be completely unresonsive, until the
+execution of the CLI handler is fully completed. Note that even
+`ceph orch ps`` will not respond, while another handler is executed.
+
+This means, we should only do very few calls to remote hosts synchronously. 
+As a guideline, cephadm should do at most ``O(1)`` networks calls in CLI handlers.
```
Comment (mgfritch) [inline doc/dev/cephadm.rst:101]:
```suggestion
As a guideline, cephadm should do at most ``O(1)`` network calls in CLI handlers. 
```

**Comment by sebastian-philipp** [comment]
jenkins retest this please

**Comment by sebastian-philipp** [comment]
jenkins test make check

**Comment by sebastian-philipp** [comment]
jenkins test make check

**Pull request merged**